src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Interop/InvokeWithJsonMarshalling.ts

```diff
@@ -1,27 +1,41 @@
 ﻿import { platform } from '../Environment';
 import { System_String } from '../Platform/Platform';
 import { getRegisteredFunction } from './RegisteredFunction';
+import { invokeDotNetMethod } from './DotNetInvoker';
 import { getElementByCaptureId } from '../Rendering/ElementReferenceCapture';
+import { MethodOptions } from './MethodOptions'
 
 const elementRefKey = '_blazorElementRef'; // Keep in sync with ElementRef.cs
 
 export function invokeWithJsonMarshalling(identifier: System_String, ...argsJson: System_String[]) {
-  const identifierJsString = platform.toJavaScriptString(identifier);
-  const funcInstance = getRegisteredFunction(identifierJsString);
-  const args = argsJson.map(json => JSON.parse(platform.toJavaScriptString(json), jsonReviver));
-  const result = funcInstance.apply(null, args);
-  if (result !== null && result !== undefined) {
-    const resultJson = JSON.stringify(result);
-    return platform.toDotNetString(resultJson);
-  } else {
-    return null;
-  }
+    const identifierJsString = platform.toJavaScriptString(identifier);
+    const funcInstance = getRegisteredFunction(identifierJsString);
+    const args = argsJson.map(json => JSON.parse(platform.toJavaScriptString(json), jsonReviver));
+    const result = funcInstance.apply(null, args);
+    if (result !== null && result !== undefined) {
+        const resultJson = JSON.stringify(result);
+        return platform.toDotNetString(resultJson);
+    } else {
+        return null;
+    }
 }
 
+export function invokeWithJsonMarshallingAsync<T>(identifier: System_String, asyncProtocol: System_String, ...argsJson: System_String[]) {
+    const identifierJsString = platform.toJavaScriptString(identifier);
+    const funcInstance = getRegisteredFunction(identifierJsString);
+    const asyncJsString = platform.toJavaScriptString(asyncProtocol);
+    const async = JSON.parse(asyncJsString) as { success: string, failure: string, function: MethodOptions };
```

Comment:
It looks like in every case, the value for `function` is identical (it encodes a reference to `InvokeTaskCallback`). Since we control the protocol we know this will always be the case. So could we remove `function` from here and avoid having to pass this information on every call?